enterprise/functions/src/main/java/io/crate/operation/aggregation/HyperLogLogDistinctAggregation.java

```diff
@@ -63,22 +62,43 @@
     }
 
     public static void register(EnterpriseFunctionsModule mod) {
-        for (DataType<?> t : DataTypes.PRIMITIVE_TYPES) {
-            mod.register(new HyperLogLogDistinctAggregation(new FunctionInfo(
-                new FunctionIdent(NAME, Collections.singletonList(t)), DataTypes.LONG,
-                FunctionInfo.Type.AGGREGATE),
-                t));
-            mod.register(new HyperLogLogDistinctAggregation(new FunctionInfo(
-                new FunctionIdent(NAME, Arrays.asList(t, DataTypes.INTEGER)), DataTypes.LONG,
-                FunctionInfo.Type.AGGREGATE),
-                t));
+        for (var supportedType : DataTypes.PRIMITIVE_TYPES) {
+            mod.register(
+                Signature.aggregate(
+                    NAME,
+                    supportedType.getTypeSignature(),
+                    DataTypes.LONG.getTypeSignature()),
+                args -> {
+                    var info = new FunctionInfo(
+                        new FunctionIdent(NAME, args),
+                        DataTypes.LONG,
+                        FunctionInfo.Type.AGGREGATE
+                    );
+                    return new HyperLogLogDistinctAggregation(info, supportedType);
+                }
+            );
+            mod.register(
+                Signature.aggregate(
+                    NAME,
+                    supportedType.getTypeSignature(),
+                    DataTypes.LONG.getTypeSignature(),
```

Comment:
I have to use here a bit wider data type because otherwise functions like `hyperloglog_distinct(ref-some-type, 18)` won't be matched.  I think it should safe because of this call https://github.com/crate/crate/pull/9867/files#diff-dfd009c6f235b400cc4d0031d384d1a2L103. Not sure, maybe we can even move up the precondition check for the precision argument from the state.init to assert that it is not out of the argument range before converting it to int.